wasmtime-wasi/src/host_impls.rs

```diff
@@ -1,5 +1,63 @@
+use super::host;
 use super::wasm32;
 
+pub unsafe fn wasmtime_ssp_args_get(
+    argv_environ: *mut host::argv_environ_values,
+    argv: *mut *mut host::char,
+    argv_buf: *mut host::char,
+) -> wasm32::__wasi_errno_t {
+    for i in 0..(*argv_environ).argc as isize {
+        let buf_off = isize::wrapping_sub(
```

Comment:
Using isize arithmetic makes sense, for the reasons you mentioned. `wrapping_sub` is interesting though, because if it ever wraps, that would mean the argv data is too big to be handled by wasm, and we should have caught such a problem earlier. I'm inclined towards a checked subtraction with a panic in overflow here; what do you think?

Reply:
Yeah, `checked_sub` is better here; I lifted `wrapping_sub` from the `offset_from` code.